oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/EventHookCallback.java

```diff
@@ -0,0 +1,153 @@
+import org.apache.skywalking.apm.network.event.v3.Source;
+import org.apache.skywalking.apm.network.event.v3.Type;
+import org.apache.skywalking.oap.server.analyzer.event.EventAnalyzerModule;
+import org.apache.skywalking.oap.server.analyzer.event.EventAnalyzerService;
+import org.apache.skywalking.oap.server.core.alarm.AlarmCallback;
+import org.apache.skywalking.oap.server.core.alarm.AlarmMessage;
+import org.apache.skywalking.oap.server.core.analysis.IDManager;
+import org.apache.skywalking.oap.server.core.source.DefaultScopeDefine;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * EventCallBack: When an alert is present, an event is generated for each alert message. These events are then sent to the internal event analyzer.
+ *
+ */
+public class EventHookCallback implements AlarmCallback {
+
+    private final ModuleManager manager;
+
+    public EventHookCallback(ModuleManager manager) {
+        this.manager = manager;
+    }
+
+    @Override
+    public void doAlarm(List<AlarmMessage> alarmMessage) {
+        EventAnalyzerService analyzerService = manager.find(EventAnalyzerModule.NAME).provider().getService(EventAnalyzerService.class);
+        alarmMessage.forEach(a -> {
+            for (Event event : constructCurrentEvent(a)) {
+                analyzerService.analyze(event);
+            }
+        });
+    }
+
+    private Event[] constructCurrentEvent(AlarmMessage msg) {
+        Event[] events = new Event[2];
+        long millis = System.currentTimeMillis();
+        Event.Builder builder = Event.newBuilder()
+                .setUuid(UUID.randomUUID().toString())
+                .setName("Alarm")
+                .setStartTime(millis - msg.getPeriod())
+                .setMessage(msg.getAlarmMessage())
+                .setType(Type.Error)
+                .setEndTime(millis);
+
+        switch (msg.getScopeId()) {
+            case DefaultScopeDefine.SERVICE :
+                IDManager.ServiceID.ServiceIDDefinition singleServiceIdDef = IDManager.ServiceID.analysisId(msg.getId0());
+                builder.setSource(
+                    Source.newBuilder()
+                            .setService(singleServiceIdDef.getName())
+                            .build()
+                );
+                events[0] = builder.build();
+                break;
+            case DefaultScopeDefine.SERVICE_RELATION :
+                IDManager.ServiceID.ServiceIDDefinition doubleServiceIdDef = IDManager.ServiceID.analysisId(msg.getId0());
+                builder.setSource(
+                        Source.newBuilder()
+                            .setService(doubleServiceIdDef.getName())
+                            .build()
+                );
+                events[0] = builder.build();
+                doubleServiceIdDef = IDManager.ServiceID.analysisId(msg.getId1());
+                builder.setSource(
```

Comment:
When building the second event, you need to reset the uuid

Reply:
Ok.